### DISCUSSION TITLE
How to abort execution, if I only need few lines to be parsed?

### DISCUSSION THREAD

**Discussion by IoBes** [airbreather/Cursively#27] [Q&A] [answered]
In my project I need to split reading of CSV in to two parts.
1. parsing of the header and stop reading of the rest of the file
2. read the rest of the data

if I use example code:
	CsvSyncInput.ForMemoryMappedFile(csvFilePath).WithDelimiter((byte)';').Process(new MyVisitor());

How can I achieve point 1.?

**Comment by airbreather** [comment, marked as answer]
Recommendation is to use [`CsvReaderVisitorWithUTF8HeadersBase`](https://airbreather.github.io/Cursively/api/Cursively.CsvReaderVisitorWithUTF8HeadersBase.html) as the base class for your visitor, since that gives you a [special method](https://airbreather.github.io/Cursively/api/Cursively.CsvReaderVisitorWithUTF8HeadersBase.html#Cursively_CsvReaderVisitorWithUTF8HeadersBase_VisitEndOfHeaderRecord) that you can override to plug in your header-specific processing.

**Reply by IoBes** [reply]
Thank you for this great hint.
After adding:
protected override void VisitEndOfHeaderRecord()
			{
				// We throw Exception to abort parsing of all other lines, because we need only headers
				if ( _processOnlyHeader ) {
					throw new Exception("Finished to read headers");
				}
			}
I was able to realize what I need
